docs/DeveloperGuide.md

```diff
@@ -610,6 +610,53 @@ CompletableFuture<List<Application>> applications =
     );
 ```
 
+## Keep a Language Server alive with a Lease
+
+If you need a reference to your language server and need to execute operations with it, 
+for an extended period of time, you can use `LanguageServerItem.keepAlive()` method to create
+a `Lease` on the item. The lease represents a request for lsp4ij not to terminate the language 
+server until the `Lease` is disposed.
+
+```java
+Project project = ...
+CompletableFuture<Lease<LanguageServerItem>> serverLease = 
+        LanguageServerManager.getInstance(project).getLanguageServer("myLanguageServerId")
+        .thenApply(item -> item.keepAlive());
+```
+
+As long as the lease has not been disposed this will request lsp4ij not arbitrarily terminate the language
+server under 'normal circumstances'. This means that the server will not be terminated for example when
+there are no more open editors.
+
+To get access to your language server from the lease call the `get()` method on the lease.
+This will return a LanuageServerItem that is guaranteed to have an 'active' server at that time.
+
+```java
+serverLease.thenAccept(lease -> {
+    // Fetch list of 'applications' using some custom protocol on 'MyLanguageServer'
+    List<Application> applications = ((MyLanguageServer)lease.get().getServer()).getApplications();   
+    ... do something with the result... 
+});

```

Comment:
As dispose method is very important I think you should add lease.dispose() at the end of your sample code

Reply:
Good idea. I did that and also worked in the handling of `ServerWasStoppedException` which may happen if server got stopped by user or crashed.